core/src/main/java/org/infinispan/notifications/cachelistener/cluster/impl/BatchingClusterEventManagerImpl.java

```diff
@@ -85,36 +95,38 @@ public void addTargets(Address address, UUID identifier, Collection<ClusterEvent
       }
 
       @Override
-      public void sendToTargets(DistributedExecutorService service) {
-         DistributedExecutionCompletionService<Void> completion = new DistributedExecutionCompletionService<>(service);
-         int syncCount = 0;
+      public void sendToTargets(ClusterExecutor executor) {
+         List<CompletableFuture<Void>> futures = new ArrayList<>();
+         TriConsumer<Address, Void, Throwable> triConsumer = (a, v, t) -> {
+           if (t != null) {
+              throw new CacheException(t);
+           }
+         };
          for (Entry<Address, TargetEvents<K, V>> entry : targets.entrySet()) {
             TargetEvents<K, V> value = entry.getValue();
             if (value.events.size() > 1) {
+               CompletableFuture<Void> future = executor.filterTargets(Collections.singleton(entry.getKey())).submitConsumer(
```

Comment:
Nitpicking: I try to split chained method calls before `.` instead of after `(`, the way IntelliJ seems to like it. Same on line 115.

Reply:
Yeah I meant to do this. Guessing I added the filter later and didn't separate the invocations by line.